macros/src/lib.rs

```diff
@@ -421,10 +421,8 @@ fn log(level: Level, ts: TokenStream) -> TokenStream {
         if #logging_enabled {
             match (#(&(#args)),*) {
                 (#(#pats),*) => {
-                    let ts = defmt::export::timestamp();
                     if let Some(mut _fmt_) = defmt::export::acquire() {
```

Comment:
We could also make the `acquire` part of shared code, by doing something like:

```rust
impl Formatter {
    pub fn start_frame(s: &Str) -> Option<Self> {
        let ts = timestamp();
        let mut fmt = acquire();
        if let Some(fmt) = &mut fmt {
            fmt.istr(s);
            fmt.leb64(ts);
        }

        fmt
    }
}
```

As a bonus, this keeps the order of operations like it was before (timestamp, then acquire).

Reply:
Presumably we'd want to move the `#[inline(never)]` to `start_frame`, since we should only have a single call site to `acquire` remaining.

Reply:
Sounds  like a good idea. What do we do with `winfo!` then? It needs to encode the header but doesn't create its own Formatter

Reply:
hmm, good question. Ideally I'd like to get rid of it. I might look into doing that tomorrow.

Reply:
I'll merge this PR as-is for now – we can always land further improvements later.